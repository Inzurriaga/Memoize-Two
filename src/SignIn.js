import React, { Component } from "react"

export default class SignIn extends Component{
    constructor(){
        super();
        this.state = {
            username: "",
            password: ""
        }
    }

    username = e => {
        this.setState({
            username: e.target.value
        })
    }
    
    password = e => {
        this.setState({
            password: e.target.value
        })
    } 

    swtichToCreateAccount = () => {
        this.props.toggle()
    }

    login = e => {
        e.preventDefault()
        let user = this.props.users.find(user => {
            return user.name === this.state.username && user.password === this.state.password
        })
       if(typeof user === "object"){
           this.props.loginToAccount(user)
       }else{
            console.log("error message")
       }
    }

    render() {
        return(
            <form className="signin-section" onSubmit={this.login}>
            <input type="button" onClick={this.swtichToCreateAccount} value="create account"></input>
                <h2>LOGIN</h2>
                <input
                onChange={this.username} type="text" placeholder="Username"></input>
                <input onChange={this.password} type="password" placeholder="Password"></input>
                <input type="submit" value="Log In"></input>
            </form>
        )
    }
}