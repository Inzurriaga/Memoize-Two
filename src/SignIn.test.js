import React from 'react';
import { shallow } from 'enzyme';
import SignIn from "./SignIn"

const mockUsers = [{name:"duy",password:"duy",completed:[1,2,5]},{name:"gabriel",password:"gabriel",completed:[1,2,5]},{name:"isaac",password:"isaac",completed:[1,2]}]
const mockToggleLoginOption = jest.fn()
const mockLoginToAccount = jest.fn()

describe("SignIn", ()=> {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SignIn toggle={mockToggleLoginOption} 
              users={mockUsers} 
              loginToAccount={mockLoginToAccount}/>
    )
  })

  it("should match a Snap shot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should have a default state", () => {
    expect(wrapper.state()).toEqual({
        username: "",
        password: ""
    })
  })

  it("should change the username state when text is type into the username input", () => {
    expect(wrapper.state("username")).toEqual("")
    wrapper.find("input").at(1).simulate("change", {target: {value: "gabriel"}})
    expect(wrapper.state("username")).toEqual("gabriel")
  })

  it("should change the password state when text is type into the password input", () => {
    expect(wrapper.state("password")).toEqual("")
    wrapper.find("input").at(2).simulate("change", {target: {value: "gabriel"}})
    expect(wrapper.state("password")).toEqual("gabriel")
  })

  it("should invoke the toggle function when button is click", () => {
    wrapper.find("input").at(0).simulate("click")
    expect(mockToggleLoginOption).toHaveBeenCalled()
  })

  it("should invoke the submit function when button is click", () => {
    wrapper.find("input").at(1).simulate("change", {target: {value: "gabriel"}})
    wrapper.find("input").at(2).simulate("change", {target: {value: "gabriel"}})
    wrapper.find("form").simulate("submit", {preventDefault: () => {}})
    expect(mockLoginToAccount).toHaveBeenCalled()
  })
})