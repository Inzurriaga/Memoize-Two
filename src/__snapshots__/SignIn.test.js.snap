// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match a Snap shot 1`] = `
<form
  className="signin-section"
  onSubmit={[Function]}
>
  <input
    onClick={[Function]}
    type="button"
    value="create account"
  />
  <h2>
    LOGIN
  </h2>
  <input
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <input
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <input
    type="submit"
    value="Log In"
  />
</form>
`;
